terminal.py

```diff
@@ -0,0 +1,57 @@
+import subprocess
+import os
+from pathlib import Path
+
+# Class to represent an instance of a terminal which is initiated
+# by a discord user via the discord bot
+class Terminal:
+
+    # Object constructor
+    def __init__(self, name, terminalPath, directoryPath=str(Path.home())):
+        # The name of the terminal instance (supplied by the user)
+        self.name = name
+
+        # The path of the actual terminal program to be run via discord bot
+        self.terminalPath = terminalPath
+        
+        # The current working directory that the user is in (starts in the home directory)
+        self.currentDirectory = directoryPath
+    
+    # Execute a bash terminal command (passed as an argument to the function) on the
+    # machine running the bot. Returns a string message which can be displayed for the user.
+    def executeCommand(self, command):
+        # Create a Popen object which will take in the command and the working
+        #   directory and run the command when communicate is called
+        self.terminalInstance = subprocess.Popen([command], stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell = True, cwd = self.currentDirectory)
+        
+        # Use communicate to run the command and get the bytes-like objects as output/error
+        stdout, stderr = self.terminalInstance.communicate()
+        
+        # Object containing data returned by the terminal after the command was executed
+        return ({
+            'terminalName': self.name,
+            'currentDirectory': self.currentDirectory,
+            'executedCommand': command,
+            'outputString': stdout.decode('utf-8'),
+            'errorString': stderr.decode('utf-8')
+        })
+
+# ~~~~~~~~~~~~~~~~ TEST CODE (for windows) ~~~~~~~~~~~~~~~~
```

Comment:
The current state has not been tested for windows, can we test the changing directory with the optional currentDirectory parameter. @Jacob-Hunt

Reply:
@joey-kilgore Can you rephrase this comment? What, specifically, do you want tested for windows? Thanks!

Reply:
There is an optional third parameter for setting the current directory when creating a terminal instance. We should test this and ensure you can have different terminal directories called when instantiating bother the Terminal class and the Popen class

Reply:
@joey-kilgore Got it! Will test this on my Windows machine and report back shortly...

Reply:
@joey-kilgore Looks like we're good:

![terminal-py_execution](https://user-images.githubusercontent.com/26048894/63218442-f385de00-c128-11e9-827b-f80757eb02e7.PNG)